### PR TITLE
Extract out 'net/storage_thread' into separate components.

### PR DESCRIPTION
**Pull request by frewsxcv**
The logic for `storage_thread.rs` (in `net` and `net_traits`) is
already isolated from the rest of the other modules in the crates.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10768)

### CODE REVIEW

**Comment by frewsxcv** [comment]
I felt it was awkward to have the `storage_traits` stuff grouped with `net`. If we're eventually going to have something like IndexedDB, maybe some of that logic could live in this `storage` crate too?  I don't feel strongly, but if it's easy to isolate this logic into a separate crate, why not?

File: components/util/write.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use rustc_serialize::Encodable;
+use rustc_serialize::json;
+use std::error::Error;
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::Path;
+
+pub fn write_json_to_file<T: Encodable>(data: &T, profile_dir: &str, filename: &str) {
```
Comment (KiChjang) [inline components/util/write.rs]: I don't quite see the reason for this move, since only net calls this function.

Comment (frewsxcv) [inline components/util/write.rs]: Storage does

Comment (frewsxcv) [inline components/util/write.rs]: https://github.com/servo/servo/pull/10768/files#diff-1738cfef9a278e047a11307ae80f78a6R75

Comment (KiChjang) [inline components/util/write.rs]: In which case, it feels more like a storage_trait function rather than a utility function, since it deals with file IO.

Comment (frewsxcv) [inline components/util/write.rs]: It's also used in `net`.

Comment (frewsxcv) [inline components/util/write.rs]: As per a discussion on IRC, this function has been moved to `storage_traits`.

**Comment by frewsxcv** [comment]
Might as well wait for https://github.com/servo/servo/pull/10768 to merge.

**Comment by frewsxcv** [comment]
Merge conflicts addressed.

**Comment by jdm** [comment]
Thanks for investigating this! I think the argument against doing this now is that the net crate builds quickly, and each new crate introduces a little bit more cognitive overhead when figuring out dependencies. Since I don't see a compelling reason to make this split right now in terms of addressing a known problem or facilitating a known improvement, I'm going to go ahead and close this.

**Comment by frewsxcv** [comment]
> each new crate introduces a little bit more cognitive overhead when figuring out dependencies

I guess? Not too convinced, but this is not that big of a deal

**Pull request closed without merging**